pallets/stream-payment/src/lib.rs

```diff
@@ -0,0 +1,477 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+use {
+    frame_support::{
+        pallet,
+        pallet_prelude::*,
+        storage::types::{StorageDoubleMap, StorageMap},
+        traits::tokens::Balance,
+        Blake2_128Concat,
+    },
+    frame_system::pallet_prelude::*,
+    parity_scale_codec::{FullCodec, MaxEncodedLen},
+    scale_info::TypeInfo,
+    sp_runtime::traits::{AtLeast32BitUnsigned, CheckedAdd, CheckedMul, CheckedSub, One, Zero},
+    sp_std::{fmt::Debug, marker::PhantomData},
+};
+
+pub use pallet::*;
+
+/// Type able to provide the current time for given unit.
+/// For each unit the returned number should monotonically increase and not
+/// overflow.
+pub trait TimeProvider<Unit, Number> {
```

Comment:
What about using the Time trait? https://github.com/paritytech/polkadot-sdk/blob/757ae372f7f07d9a7a4f62f646baa5e388ec5531/substrate/frame/support/src/traits/misc.rs#L723

Reply:
With `Time` trait the user can't choose between differents units.

Reply:
I thought the units would be more of Mulltiples of the same unit. I think we can use seconds and I guess the rest can be handled by a dapp.

Reply:
Ok I see that the expectation is to make an enum, looks good!